lldb/source/Symbol/Type.cpp

```diff
@@ -145,10 +159,16 @@ bool TypeQuery::ContextMatches(
     ++pat;
   }
 
-  // Skip over any remaining module entries if we were asked to do that.
-  while (GetIgnoreModules() && ctx != ctx_end &&
-         ctx->kind == CompilerContextKind::Module)
-    ++ctx;
+  // Skip over any remaining module and anonymous namespace entries if we were
+  // asked to do that.
+  auto should_skip = [this](const CompilerContext &ctx) {
+    if (ctx.kind == CompilerContextKind::Module)
+      return GetIgnoreModules();
+    if (ctx.kind == CompilerContextKind::Namespace && ctx.name.IsEmpty())
```

Comment:
Almost wonder if this check should be a `IsAnonymousNamespace` helper function on `CompilerContext`. But maybe that'd make it too C++-specific? Since it's just in this function, feel free to ignore

Reply:
In the first version of the patch I had an `IsAnonymousNamespace` and `MatchesAnonymousNamespace` (the difference being in how they test the `kind` bits) . This was before I changed everything to use an empty string, after which it did not seem that useful...